app/src/main/java/org/wikipedia/notifications/NotificationFilterItemView.kt

```diff
@@ -0,0 +1,67 @@
+package org.wikipedia.notifications
+
+import android.content.Context
+import android.os.Build
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.LinearLayout
+import androidx.appcompat.content.res.AppCompatResources
+import org.wikipedia.R
+import org.wikipedia.databinding.ItemNotificationFilterBinding
+import org.wikipedia.search.SearchFragment
+import org.wikipedia.util.DeviceUtil
+import org.wikipedia.util.ResourceUtil
+import org.wikipedia.views.ViewUtil
+
+class NotificationFilterItemView constructor(context: Context, attrs: AttributeSet? = null) : LinearLayout(context, attrs) {
+
+    interface Callback {
+        fun onCheckedChanged(langCode: String)
+    }
+
+    private var binding = ItemNotificationFilterBinding.inflate(LayoutInflater.from(context), this)
+    var callback: Callback? = null
+
+    init {
+        layoutParams = ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, 200)
+        setBackgroundColor(ResourceUtil.getThemedColor(context, R.attr.paper_color))
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            foreground =
+                AppCompatResources.getDrawable(context, ResourceUtil.getThemedAttributeId(context, R.attr.selectableItemBackground))
+        }
+        setOnClickListener {
+            when {
+                binding.notificationFilterTitle.text.toString() == context.getString(R.string.wikimedia_commons) -> {
+                    callback?.onCheckedChanged("commons")
+                }
+                binding.notificationFilterTitle.text.toString() == context.getString(R.string.wikidata) -> {
+                    callback?.onCheckedChanged("wikidata")
+                }
+                else -> {
+                    callback?.onCheckedChanged(binding.notificationFilterLanguageCode.text!!.toString())
+                }
```

Comment:
It can be simplified as following:

```
when (binding.notificationFilterTitle.text.toString()) {
                context.getString(R.string.wikimedia_commons) -> callback?.onCheckedChanged("commons")
                context.getString(R.string.wikidata) -> callback?.onCheckedChanged("wikidata")
                else -> callback?.onCheckedChanged(binding.notificationFilterLanguageCode.text!!.toString())
            }
```